react/src/components/Table/Table.tsx

```diff
@@ -512,7 +507,32 @@ const Table: React.FC<TableProps> = ({ variantData }) => {
                                                                 <Flex
                                                                     alignItems="center"
                                                                     justifyContent="center"
+                                                                    nowrap
                                                                 >
+                                                                    {/* 
+                                                                        Filter icon for filtering individual columns
+                                                                    */}
+                                                                    {!column.disableFilters && (
+                                                                        <FilterPopover
+                                                                            columns={visibleColumns.filter(
+                                                                                c =>
+                                                                                    c.id ===
+                                                                                    column.id
+                                                                            )}
```

Comment:
I think filtering through `visibleColumns` is redundant because we check if the column is redundant at line 486. 

Also, `columns` property for this `FilterPopoever` component will always only contain 1 column whereas its type is set to be an array of columns. Maybe you can check if `columns` property can be reduced to 1 column without impacting other components.